src/lightning/pytorch/loops/evaluation_loop.py

```diff
@@ -446,9 +456,16 @@ def _verify_dataloader_idx_requirement(self) -> None:
         step_hook = "test_step" if trainer.testing else "validation_step"
         batch_start_hook = "on_test_batch_start" if trainer.testing else "on_validation_batch_start"
         batch_end_hook = "on_test_batch_end" if trainer.testing else "on_validation_batch_end"
-        assert self._combined_loader is not None
         _verify_dataloader_idx_requirement(
-            (step_hook, batch_start_hook, batch_end_hook),
+            (step_hook,),
+            self._is_sequential
+            and self.num_dataloaders > 1
+            and not isinstance(self._data_fetcher, _DataLoaderIterDataFetcher),
+            self._stage,
+            trainer.lightning_module,
+        )
+        _verify_dataloader_idx_requirement(
```

Comment:
I don't get why there are two `_verify_dataloader_idx_requirement` calls after each other

Reply:
The verification is done for different hooks. `validation_step` will not take dataloader_idx if using `dataloader_iter` but `on_validation_batch_start,end` will